### PR TITLE
Use pytest in tests of grf simulate

### PR DESCRIPTION
**Pull request by dafeda**

### CODE REVIEW

**Review by sindre-nistad — CHANGES_REQUESTED** [review]
Looks good, but it seems to be missing a couple of `assert`s

File: tests/test_simulate.py [hunk 1]
```diff
@@ -1,37 +1,30 @@
...
-        self.assertEqual(len(a), 2)
-        self.assertEqual(a[0], 406)
-        self.assertEqual(a[1], 741)
+def test_simulation_size_3d_equal():
+    v = grf.variogram('spherical', 1000.0, 1000.0, 1000.0)
+    a = grf.simulation_size(v, 100, 10.0, 100, 10.0, 100, 10.0)
+    len(a) == 3
+    a[0] == 201
+    a[1] == 201
+    a[2] == 201
```
[the remark this answers is not in the document]
Comment (dafeda) [reply tests/test_simulate.py]: Fixed, thanks 👍

File: tests/test_simulate.py [hunk 2]
```diff
@@ -1,37 +1,30 @@
...
-
-if __name__ == '__main__':
-    unittest.main()
+def test_simulation_size_3d_uneven():
+    v = grf.variogram('spherical', 1000.0, 250.0, 125.0)
+    a = grf.simulation_size(v, 507, 50.0, 305, 17.0, 103, 4.0)
+    len(a) == 3
+    a[0] == 528
+    a[1] == 320
+    a[2] == 135
```
[the remark this answers is not in the document]
Comment (dafeda) [reply tests/test_simulate.py]: Fixed, thanks 👍

File: tests/test_simulate.py [hunk 3]
```diff
@@ -1,37 +1,30 @@
-import unittest
 import gaussianfft as grf
 
+def test_simulation_size_1d():
+    v = grf.variogram('spherical', 100.0)
+    a = grf.simulation_size(v, 100, 1.0)
+    assert len(a) == 1
+    a[0] == 201
```
Comment (sindre-nistad) [inline tests/test_simulate.py:7]:
```suggestion
    assert a[0] == 201
```
missing `assert`?

File: tests/test_simulate.py [hunk 4]
```diff
@@ -1,37 +1,30 @@
...
-        self.assertEqual(a[1], 320)
-        self.assertEqual(a[2], 135)
-
-
-if __name__ == '__main__':
-    unittest.main()
+def test_simulation_size_3d_uneven():
+    v = grf.variogram('spherical', 1000.0, 250.0, 125.0)
+    a = grf.simulation_size(v, 507, 50.0, 305, 17.0, 103, 4.0)
+    len(a) == 3
```
Comment (sindre-nistad) [inline tests/test_simulate.py:27]:
```suggestion
    assert len(a) == 3
```

missing assert?

File: tests/test_simulate3d.py [hunk 5]
```diff
@@ -1,66 +1,58 @@
...
-            np.count_nonzero(amax != int(self.ny/2 - 1)),
-            amax.size * 0.8
-        )
-        amax = np.argmax(np.abs(diffs2), axis=2)
-        self.assertLess(
-            np.count_nonzero(amax != int(self.nz/2 - 1)),
-            amax.size * 0.8
-        )
+    np.std(diffs0) < np.std(diffs1)
+    np.std(diffs1) < np.std(diffs2)
```
Comment (sindre-nistad) [inline tests/test_simulate3d.py:34]:
```suggestion
    assert np.std(diffs0) < np.std(diffs1)
    assert np.std(diffs1) < np.std(diffs2)
```
missing `asserts`?

**Review by sindre-nistad — CHANGES_REQUESTED** [review]
There seem to be a couple of missing asserts

File: tests/test_simulate.py (hunk 3 above)
Comment (dafeda) [reply tests/test_simulate.py:7]: Thanks, I have added the assert.

**Review by sindre-nistad — CHANGES_REQUESTED** [review]
There may be more missing `assert`s. Please have a look for them

**Comment by sindre-nistad** [comment]
@dafeda; once approved, please preform `git rebase main` and squash the commits down to one

**Comment by dafeda** [comment]
Think I've included all asserts.
Have squashed.

**Pull request merged**